### ISSUE TITLE
RFE - No way to change visibility of CanvasLayer.

### ISSUE THREAD

**Issue by Sslaxx** [godotengine/godot#16532]
**Godot version:**
3.0 (but _long_ before that version).

**OS/device including version:** 
N/A

**Issue description:**

`CanvasLayer` nodes lack `show` or `hide` functions or `visible` properties, meaning that you cannot hide the node itself and instead either have to rely on the `layer` property, or invoke show/hide/visible on each item within the CanvasLayer should they have them. And whilst `layer` should handle most if not all cases, some projects may have layer actions that might make handling visiblity properties more desirable.

**Steps to reproduce:**
Add CanvasLayer, try to use `show ()` or `hide ()` or `visible` with it.

**Comment by girng** [comment]
just do

Canvas Layer -> MainNode
- Item1, Item2,  Item3, Item4, Item,5
- Item6 -> SubItem1, Subitem2, etc

then just call `hide` on `MainNode.hide()` now all items and subitems will be hidden could be regular control or panel

**Comment by QbieShay** [comment]
This feature is still missing and I think it is very needed. I'll present you my use case:

I have a very very complex interface and establishing who stays in front of who via the order in the node hierarchy has proved itself very fragile. There are also some cases where you want only a nested part of one of the component to be on top of everything. 
I think canvas layers excels in this, they make the design of the overall UI much more reliable and flexible. Also, they allow for easier refactoring since you don't rely on the hierarchy anymore.
Allowing them to show/hide is great for both the running software, which grants more control, and for debugging purposes.

**Comment by williamd1k0** [comment]
For now you could use `CanvasLayer.scale` as a workaround:

```gdscript
self.scale = Vector2(0, 0) # Hide
self.scale = Vector2(1, 1) # Show
```

**Comment by TheRealCatherine** [comment]
Using the scaling method mentioned above is giving me the following error (a ton of times) when setting the scale to 0:

E 0:00:08:0030   Condition ' det == 0 ' is true.
  <C Source>     core/math/transform_2d.cpp:51 @ affine_invert()

to reproduce:
clone https://gitlab.com/youreperfectstudio/catherines-quest.git
use commit e073b920699d179f37bbceae3268fef1faec9c4b
click through the title and character selection screens
on the character dialog screen press Close button check console
its called from story/engine/DialogController.gd in the show/hide _workaround functions

**Comment by williamd1k0** [comment]
>Using the scaling method mentioned above is giving me the following error (a ton of times) when setting the scale to 0:
>
>E 0:00:08:0030 Condition ' det == 0 ' is true.
>core/math/transform_2d.cpp:51 @ affine_invert()

I can confirm that these errors are appearing.
I used to use this method in godot 2.1 with no issues, though.

**Comment by hsandt** [comment]
This does not only affect scripting, but the editor too. Since there is no visibility property, there is no "eye" icon to Toggle Visibility of a Canvas Layer in the Scene. As ghost mentioned, you can put an intermediate Control node just under the Canvas layer to leverage this. But if your Canvas Layer is an instanced Scene, you won't see the Control since only the top-level of instanced Scenes is visible by default.

![image](https://user-images.githubusercontent.com/1067772/73883751-652f2100-485c-11ea-9983-f7d1673f04b6.png)

As a workaround, you can check Editable Children to make children appear, including the Control node. Unfold the Canvas Layer instance, but fold everything under the Control node so it doesn't take too much place in the Scene tree. Now you have this:

![image](https://user-images.githubusercontent.com/1067772/73883977-d40c7a00-485c-11ea-9845-fee4b98460d2.png)

And you can toggle visibility by clicking on the eye icon on the Control node.

Note that it requires an extra node and to keep Editable Children checked.

**Comment by kapes232** [comment]
Hi all, this "issue" is same for me, i solve it by putting all i want to hide in a node and then removing this node from the tree then putting it again in tree, it works for me.

**Comment by Calinou** [comment]
Feature and improvement proposals for the Godot Engine are now being discussed and reviewed in a dedicated [**Godot Improvement Proposals (GIP)**](https://github.com/godotengine/godot-proposals) ([godotengine/godot-proposals](https://github.com/godotengine/godot-proposals)) issue tracker. The GIP tracker has a detailed issue template designed so that proposals include all the relevant information to start a productive discussion and help the community assess the validity of the proposal for the engine.

The main ([godotengine/godot](https://github.com/godotengine/godot)) tracker is now solely dedicated to bug reports and Pull Requests, enabling contributors to have a better focus on bug fixing work. Therefore, we are now closing all older feature proposals on the main issue tracker.

If you are interested in this feature proposal, please [**open a new proposal on the GIP tracker**](https://github.com/godotengine/godot-proposals/issues/new/choose) following the given issue template (after checking that it doesn't exist already). Be sure to reference this closed issue if it includes any relevant discussion (which you are also encouraged to summarize in the new proposal). Thanks in advance!

**Comment by Sslaxx** [comment]
Fixed by #48006
